tests/components/reolink/test_button.py

```diff
@@ -0,0 +1,111 @@
+"""Test the Reolink button platform."""
+
+from unittest.mock import MagicMock, patch
+
+import pytest
+from reolink_aio.exceptions import ReolinkError
+
+from homeassistant.components.button import DOMAIN as BUTTON_DOMAIN
+from homeassistant.components.reolink import const
+from homeassistant.components.reolink.button import ATTR_SPEED
+from homeassistant.config_entries import ConfigEntryState
+from homeassistant.const import ATTR_ENTITY_ID, Platform
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import HomeAssistantError
+from homeassistant.helpers import entity_registry as er
+
+from .conftest import TEST_NVR_NAME, TEST_UID
+
+from tests.common import MockConfigEntry
+
+
+async def test_button(
+    hass: HomeAssistant,
+    config_entry: MockConfigEntry,
+    reolink_connect: MagicMock,
+) -> None:
+    """Test button entity with ptz up."""
+    with patch("homeassistant.components.reolink.PLATFORMS", [Platform.BUTTON]):
+        assert await hass.config_entries.async_setup(config_entry.entry_id)
+    await hass.async_block_till_done()
+    assert config_entry.state is ConfigEntryState.LOADED
+
+    entity_id = f"{Platform.BUTTON}.{TEST_NVR_NAME}_ptz_up"
+
+    await hass.services.async_call(
```

Comment:
This isn't a button though?

Reply:
What do you mean?
The ptz_up is indeed a button entity.

Reply:
The "ptz_move" service is a "entity service", so it is linked to the PTZ button entities.
It does the same as pressing the button entity but allows you to specify a additional "speed" attribute for the speed the camera will move.

Reply:
The code for the ptz_move service is in the button.py file since it is an entity service so I also put the tests in the test_button.py file
Do you want me to split out the tests of the "prz_move" service to a seperate test?
And do you want this test in test_button.py or in the test_services.py?

Reply:
I now split out the test, but kept it in the test_button.py file.